### PR TITLE
Pygit2 recipe

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

**Comment by kszucs** [comment]
Pygit2 needs an environment variable `LIBGIT2=<conda-dir>/lib` . How can I access the conda installation path?

File: recipes/pygit2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+test:
+  # Python imports
+  imports:
+    - pygit2
+
+  # commands:
+    # You can put test commands to be run here.  Use this to test that the
+    # entry points work.
+
```
Comment (shadowwalkersb) [inline recipes/pygit2/meta.yaml]: Add `python -c 'import pygit2'` under `commands`.

**Comment by shadowwalkersb** [comment]
> Pygit2 needs an environment variable LIBGIT2=<conda-dir>/lib . How can I access the conda installation path?

I believe something like this should get you going.
Non-Windows: `export LIBGIT2=${PREFIX}/lib`
Windows: `set LIBGIT2=%LIBRARY_LIB%`

See http://conda.pydata.org/docs/building/environment-vars.html for more details.

**Comment by shadowwalkersb** [comment]
or according to the documentation it may have to be 
`export LIBGIT2=${PREFIX}` and `set LIBGIT2=%LIBRARY%`

File: recipes/pygit2/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    - setuptools
+    - cffi
+    - six
+    - libgit2
+
+  run:
+    - python
+    - cffi
+    - six
+    - libgit2
```
Comment (shadowwalkersb) [inline recipes/pygit2/meta.yaml]: Again, pin here as above.

File: recipes/pygit2/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+build:
+  number: 0
+  features:
+    - vc9    # [win and py27]
+    - vc10   # [win and py34]
+    - vc14   # [win and py35]
+
+requirements:
+  build:
```
Comment (shadowwalkersb) [inline recipes/pygit2/meta.yaml]: Also, add this to the build requirements (I guess it is to be prepared for some future changes):

```
    - vc 9    # [win and py27]
    - vc 10   # [win and py34]
    - vc 14   # [win and py35]
```

**Comment by shadowwalkersb** [comment]
@kszucs , after you are done with the suggested changes, I suggest cleaning up the commented lines. After that, if the tests pass, I think, you can ask for a review.

**Comment by kszucs** [comment]
@shadowwalkersb Please review!

**Comment by shadowwalkersb** [comment]
@conda-forge/staged-recipes , we need a review here. Also, CircleCI doesn't seem to build anything. Any suggestions?

File: recipes/pygit2/bld.bat [hunk 4]
```diff
@@ -0,0 +1,3 @@
+set LIBGIT2=%LIBRARY%
```
Comment (jakirkham) [inline recipes/pygit2/bld.bat]: Should this be quoted?

File: recipes/pygit2/meta.yaml (hunk 3 above)
Comment (jakirkham) [inline recipes/pygit2/meta.yaml]: That shouldn't be needed. `python` already does this for us.

File: recipes/pygit2/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    - vc 9    # [win and py27]
+    - vc 10   # [win and py34]
+    - vc 14   # [win and py35]
```
Comment (jakirkham) [inline recipes/pygit2/meta.yaml]: Please drop this.

File: recipes/pygit2/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    - python
+    - setuptools
+    - cffi
+    - six
```
Comment (jakirkham) [inline recipes/pygit2/meta.yaml]: It doesn't look like `six` is required for the build. Could we please drop it if that is the case?

File: recipes/pygit2/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    - libgit2 {{version_major}}.{{version_minor}}.*
+
+  run:
+    - python
+    - cffi
+    - six
+    - libgit2 {{version_major}}.{{version_minor}}.*
+
+test:
+  # Python imports
```
Comment (jakirkham) [inline recipes/pygit2/meta.yaml]: Please drop this comment.

File: recipes/pygit2/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+  imports:
+    - pygit2
+
+  commands:
+    - python -c "import pygit2"
```
Comment (jakirkham) [inline recipes/pygit2/meta.yaml]: This is identical to the `import` test above. If we aren't going to do some more testing here, can we please drop this `command` test?

File: recipes/pygit2/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+about:
+  home: http://github.com/libgit2/pygit2
+  license: GPLv2 with linking exception
+  summary: 'Python bindings for libgit2.'
+  license_family: GPL2
```
Comment (jakirkham) [inline recipes/pygit2/meta.yaml]: Please add `license_file: COPYING` to included the license in the package.

File: recipes/pygit2/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+  doc_url: http://www.pygit2.org/
+  dev_url: https://github.com/libgit2/pygit2
+
+extra:
+  recipe-maintainers:
+    # GitHub IDs for maintainers of the recipe.
```
Comment (jakirkham) [inline recipes/pygit2/meta.yaml]: Please drop this comment.

File: recipes/pygit2/bld.bat [hunk 11]
```diff
@@ -0,0 +1,3 @@
...
+"%PYTHON%" setup.py install
```
Comment (jakirkham) [inline recipes/pygit2/bld.bat]: Please add the options `--single-version-externally-managed --record record.txt` after `install`.

File: recipes/pygit2/build.sh [hunk 12]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+export LIBGIT2=${PREFIX}
+
+$PYTHON setup.py install
```
Comment (jakirkham) [inline recipes/pygit2/build.sh]: Please add the options `--single-version-externally-managed --record record.txt` after `install`.

**Comment by jakirkham** [comment]
Thanks for the contribution. Some simple fixes mentioned above to get this cleaned up. Shouldn't take too long.

**Comment by kszucs** [comment]
@conda-forge/staged-recipes please review

**Comment by scopatz** [comment]
LGTM, thanks!

**Pull request merged**